core/templates/local_vector.h

```diff
@@ -137,10 +135,16 @@ class LocalVector {
 	}
 	_FORCE_INLINE_ bool is_empty() const { return count == 0; }
 	_FORCE_INLINE_ U get_capacity() const { return capacity; }
-	_FORCE_INLINE_ void reserve(U p_size) {
-		p_size = tight ? p_size : nearest_power_of_2_templated(p_size);
+	void reserve(U p_size) {
 		if (p_size > capacity) {
-			capacity = p_size;
+			if (tight) {
+				capacity = p_size;
+			} else {
+				capacity = MAX((U)2, capacity + ((1 + capacity) >> 1));
```

Comment:
This could use a comment that explains this is just calculates `capacity * 1.5`.